odl/discr/lp_discr.py

```diff
@@ -387,14 +385,9 @@ def __repr__(self):
             use_uniform = False
 
         if use_uniform:
-            if isinstance(self.dspace, Fn):
-                impl = 'numpy'
-                default_dtype = np.float64
-            elif isinstance(self.dspace, CudaFn):
-                impl = 'cuda'
-                default_dtype = np.float32
-            else:  # This should never happen
-                raise RuntimeError('unable to determine data space impl')
+            impl = self.dspace.impl
+            default_dtype = self.dspace.default_dtype(RealNumbers())
```

Comment:
If we at some point add that then sure, for now we dont have that problem. The more I think about that one the more sceptical I am as-well. It makes the code more indirect and harder to follow.